core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/AmazonSyncHttpClient.java

```diff
@@ -106,6 +108,16 @@ public interface RequestExecutionBuilder {
          */
         RequestExecutionBuilder executionContext(ExecutionContext executionContext);
 
+        RequestExecutionBuilder httpClientDependencies(HttpClientDependencies httpClientDependencies);
+
+        HttpClientDependencies httpClientDependencies();
```

Comment:
Do we use this getter? I don't see getters on the other methods, so suggest avoiding it if we don't use it. If we need it, I guess it doesn't hurt.

Reply:
We don't use it right now, I will remove it.

Reply:
Sorry, yes, we do use it here:

```java
        default RequestExecutionBuilder httpClientDependencies(Consumer<HttpClientDependencies.Builder> mutator) {
            HttpClientDependencies.Builder builder = httpClientDependencies().toBuilder();
            mutator.accept(builder);
            return httpClientDependencies(builder.build());
        }
```

I could remove this method and all the needed work outside but this feels a lot easier to implement, let me know if you feel like removing it, and I can look at making the changes.